### ISSUE TITLE
Use new regex support on Azure SQL

### ISSUE THREAD

**Issue by ErikEJ** [efcore/EFCore.CheckConstraints#156]
https://techcommunity.microsoft.com/t5/azure-sql-blog/introducing-regular-expression-regex-support-in-azure-sql-db/ba-p/4072741

**Comment by roji** [comment]
Nice! This seems to be in private beta, let's look at this when it releases... I hope on-premise SQL Server gets it too.

**Comment by ErikEJ** [comment]
I have signed up for the preview.

Thoughts on how to enable this feature?

1. Detect that we are on Azure SQL DB based on the server name suffix.

2. Add an option to the UseCheckconstraints method to opt in?

3. Something else?

**Comment by roji** [comment]
Yeah, EF generally avoids trying to auto-detect database version/types... We're not sure that server suffix-based detection is reliable, and in some situations we may not even have a connection string when we need to do stuff.

Here's one thought... Version 8.0 introduced UseAzureSqlDefaults() on the SqlServer options, so the plugin *should* be able to get the options and check whether that opt-in has been set... Otherwise of course we can have config in the plugin itself as you suggest...

**Comment by ErikEJ** [comment]
Finally got in to the preview, so I will start experimenting soon

**Comment by ErikEJ** [comment]
https://www.mssqltips.com/sqlservertip/8187/basic-regex-emulator-for-sql-server/?utm_source=rss&utm_medium=rss&utm_campaign=basic-regex-emulator-for-sql-server
